components/inputs/input-number.js

```diff
@@ -45,9 +62,24 @@ class InputNumber extends LitElement {
 		this.labelHidden = false;
 		this.required = false;
 
+		this._formattedValue = '';
 		this._inputId = getUniqueId();
 	}
 
+	get value() { return this._value; }
+	set value(val) {
+		const oldValue = this.value;
+		try {
+			const newValue = checkMinMaxValue(val, this.min, this.max);
+			this._formattedValue = formatValue(newValue, this.minFractionDigits, this.maxFractionDigits);
+			this._value = parseNumber(this._formattedValue);
+		} catch (err) {
+			this._formattedValue = '';
+			this._value = undefined;
```

Comment:
If theres an error do we want to wipe out the value or revert back to `oldVal`? Or maybe even allow it and display a type mismatch error?

Reply:
Is type mismatch error something we can do in form validation? If yes, then I think that's the better route to go and I'll include that in my next PR (form validation stuff).

Reply:
If we want to display an error then it'd be done as part of integrating form validation. If you take a native number input and enter something like `+--33.....` then call `input.validity` you'll see the `badInput` flag is set to `true`.

But from playing around with native inputs it looks like they don't even allow you to type non-numeric characters so there isn't an exact parallel to what we're doing here.

Reply:
So I don't think this numeric input would be exactly like the native number input because we would allow formatted numbers to be typed in as well based on locale (i.e. 3,400.23 in `en` or 30,40 in `fr`). However, throwing a validation error in this case would be better since the user may not notice that their input was changed to the "old value". I'm assuming we want an error like "Invalid value" or something when the user focuses out of the input field.